Extension/src/LanguageServer/ui_new.ts

```diff
@@ -0,0 +1,548 @@
+            };
+            return;
+        }
+
+        this.intelliSenseStatusBarItem.busy = val;
+
+        if (this.flameTimeout) {
+            clearTimeout(this.flameTimeout);
+        }
+
+        if (val) {
+            this.intelliSenseStatusBarItem.text = "$(flame)";
+            this.intelliSenseStatusBarItem.detail = this.updatingIntelliSenseText;
+        } else {
+            this.flameTimeout = setTimeout(() => {
+                if (this.intelliSenseStatusBarItem) {
+                    this.intelliSenseStatusBarItem.text = this.idleIntelliSenseText;
+                    this.intelliSenseStatusBarItem.detail = "";
+                }
+            }, this.iconDelayTime);
+        }
+        this.intelliSenseStatusBarItem.command = {
+            command: "C_Cpp.RestartIntelliSenseForFileUI_Telemetry",
+            title: localize("rescan.intellisense.text", "Rescan"),
+            tooltip: localize("rescan.intellisense.tooltip", "Rescan IntelliSense")
+        };
+    }
+
+    private codeAnalysisCurrentMode(): string {
+        const settings: CppSettings = new CppSettings((vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0) ? vscode.workspace.workspaceFolders[0]?.uri : undefined);
+        const state: string = (settings.codeAnalysisRunAutomatically && settings.clangTidyEnabled)
+            ? localize("mode.codeanalysis.status", "Automatic")
+            : localize("mode.codeanalysis.status", "Manual");
+        return state;
+    }
+
+    private setIsRunningCodeAnalysis(val: boolean): void {
+        if (this.isRunningCodeAnalysis && !val) {
+            this.codeAnalysisTotal = 0;
+            this.codeAnalysisProcessed = 0;
+        }
+        this.isRunningCodeAnalysis = val;
+        this.codeAnalysisStatusBarItem.busy = val;
+        const activeText: string = this.isCodeAnalysisPaused ? this.codeAnalysisPausedText : this.codeAnalysisRunningText;
+        const idleText: string = this.codeAnalysisModePrefix + this.codeAnalysisCurrentMode();
+        this.codeAnalysisStatusBarItem.text = val ? activeText : idleText;
+        this.codeAnalysisStatusBarItem.command = val ? {
+            command: "C_Cpp.ShowActiveCodeAnalysisCommandsUI_Telemetry",
+            title: localize("c.cpp.codeanalysis.statusbar.showCodeAnalysisOptions", "Options"),
+            // Make sure not to overwrite current progress
+            tooltip: this.codeAnalysisStatusBarItem.command?.tooltip ?? localize("startup.codeanalysis.status", "Starting...")
+        } : {
+            command: "C_Cpp.ShowIdleCodeAnalysisCommandsUI_Telemetry",
+            title: localize("c.cpp.codeanalysis.statusbar.showRunNowOptions", "Run Now")
+        };
+    }
+
+    private updateCodeAnalysisTooltip(): void {
+        this.codeAnalysProgress = localize({ key: "running.analysis.processed.tooltip", comment: [this.codeAnalysisTranslationHint] }, "Running {0}: {1} / {2} ({3}%)", this.codeAnalysisProgram,
```

Comment:
Are you certain that it works to specifying the hint/comment using a variable here?  I know that the localizable string value itself MUST be a literal in calls to `localize`, due to how values are scraped from the code.  Could you run `yarn run translations-export` and check the generated files to ensure these hints are being captured correctly?

Reply:
Great catch, i didn't notice. But this code is taken directly from the old ui implementation so it should work if it worked before.